engine/spec/components/citizens_advice_components/sidebar_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
```

Comment:
I've made quite a lot of changes to this file, please let me know if the code is as expected (I've tested locally) or if something needs adjusting/adding

Reply:
These look good.

I think you should also test for when:
- no section title url is passed in, and assert that no link is rendered in the title
- if no links are passed, nothing is rendered